lbrynet/lbrynet_daemon/Daemon.py

```diff
@@ -2174,6 +2174,27 @@ def jsonrpc_configure_ui(self, p):
         return d
 
     @AuthJSONRPCServer.auth_required
+    def jsonrpc_open_file(self, p):
+        """
+        Instruct the OS to open a file.
+
+        Args:
+            'path': path of file to be opened
+        Returns:
+            True, opens file
+        """
+        path = p['path']
+        if sys.platform == 'darwin':
+            d = threads.deferToThread(subprocess.Popen, ['open', path])
+        elif os.name == 'posix':
+            d = threads.deferToThread(subprocess.Popen, ['xdg-open', path])
+        elif sys.platform == 'win32':
+            d = threads.deferToThread(os.startfile, path)
```

Comment:
Job, Jeremy and me have kicked this around and we decided to split the difference: create a new module called something like file_utils with functions revea() and open() that have separate switch statements. A lot simpler than making three classes and a factory but still abstracts out the OS handling stuff.